client/extensions/woocommerce/app/products/product-update.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Main from 'components/main';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { editProduct, editProductAttribute, createProductActionList } from 'woocommerce/state/ui/products/actions';
+import { editProductCategory } from 'woocommerce/state/ui/product-categories/actions';
+import { getActionList } from 'woocommerce/state/action-list/selectors';
+import { getCurrentlyEditingProduct } from 'woocommerce/state/ui/products/selectors';
+import { getProductVariationsWithLocalEdits } from 'woocommerce/state/ui/products/variations/selectors';
+import { editProductVariation } from 'woocommerce/state/ui/products/variations/actions';
+import { fetchProduct } from 'woocommerce/state/sites/products/actions';
+import { fetchProductVariations } from 'woocommerce/state/sites/product-variations/actions';
+import { fetchProductCategories } from 'woocommerce/state/sites/product-categories/actions';
+import { getProductCategoriesWithLocalEdits } from 'woocommerce/state/ui/product-categories/selectors';
+import { createProduct } from 'woocommerce/state/sites/products/actions';
```

Comment:
You are importing `fetchProduct` above from the same actions file -- so these can be combined.